### DISCUSSION TITLE
What are the steps to refresh the previously selected model?

### DISCUSSION THREAD

**Discussion by Snowcatcat** [facebookexperimental/Robyn#337] [Q&A]
Can someone let me know the steps to refresh a previously selected model? I ran screenshot 1 first, but I am getting errors.
![image](https://user-images.githubusercontent.com/99045100/157160070-233b0266-d821-45e9-a68e-85c5ae0a81f6.png)

I am getting this error: object 'parallel_processing' not found

![image](https://user-images.githubusercontent.com/99045100/157160272-9a0a6f8d-1162-493f-8efb-4d3d3eaebe23.png)

I also see this piece of the code, and should I run this after I run the code in screenshot 1?
![image](https://user-images.githubusercontent.com/99045100/157160178-77068247-65b4-4ac6-9595-768915a84053.png)

**Comment by kyletgoldberg** [comment]
If you have the Robyn (RDS) object saved it will be easiest to use the code in screenshot 3. You don't need to run the code in screenshot 1 prior to running screenshot 3 either, screenshot 1's code is just for the case that you don't have the RDS object.

**Reply by Snowcatcat** [reply]
Thanks for your response @kyletgoldberg ! I'm wondering if you may help with the following example:

Let's say, I run the model for the first time and pick a winner that I like, (making stuff up) 1_181_1.
After a while, I open my R script again, if I run Robyn_Refresh (screenshot 3) directly, is it going to work? Does R know which model to work with (1_181_1)?

**Reply by laresbernardo** [reply]
Hey @Snowcatcat yes. Actually, when you export your results into a RDS file you only export your "1_181_1" model, that's why in `robyn_save()` you must specify which is your selected model (`select_model`)

**Reply by Snowcatcat** [reply]
Thanks for your reply @laresbernardo ! What if I have the results of multiple model runs in the same RDS object/folder? Like this:
![image](https://user-images.githubusercontent.com/99045100/157375679-212f048a-d7f5-448a-9542-a6f8c55afc2b.png)


I know we have set a seed number in the code, but every time I run the code, it creates a new folder with different one-pagers, and the results are different. Is it expected?

**Reply by kyletgoldberg** [reply]
That is expected yes, this is not the RDS object though. To generate that you need to run the robyn_save function which will force you to select one model ID. All of these folders are generated by the robyn_outputs. In the demo.R file, check the portion of the code with robyn_save which is what saves down the .RDS objects for future use.

**Reply by Snowcatcat** [reply]
Thanks @kyletgoldberg . This is the code I run:
![image](https://user-images.githubusercontent.com/99045100/157476546-e66aabea-17a8-4a44-8095-08e0be370a68.png)

But if I close out R, and open R again and rerun the code, it will just produce a new init folder with different model results. The Robyn_save code above will return an error.

**Reply by laresbernardo** [reply]
In the last section of the demo.R file you'll see an example to import your previous hyper-parameters, train the model and generate that OutputCollect object. The InputCollect object should be the same you ran when you first ran the code.
```
dt_hyper_fixed <- data.table::fread("~/Desktop/2022-02-21 11.29 rf11/pareto_hyperparameters.csv")
select_model <- "1_51_11" # your selected model ID
dt_hyper_fixed <- dt_hyper_fixed[solID == select_model]

OutputCollect <- robyn_run(
  # InputCollect must be provided by robyn_inputs with same dataset and parameters as before
  InputCollect = InputCollect,
  dt_hyper_fixed = dt_hyper_fixed)
```

**Reply by Snowcatcat** [reply]
Thanks! I followed the steps that you showed above and getting this error. Could you let me know what I possibly did wrong?
![image](https://user-images.githubusercontent.com/99045100/157585189-b286359e-b53a-4fcd-b2d1-c347e664e134.png)

**Reply by kyletgoldberg** [reply]
Does your dt_hyper_fixed object have a lambda hyperparameter?
